lib/wanda_web/schemas/execution_response.ex

```diff
@@ -0,0 +1,58 @@
+defmodule WandaWeb.Schemas.ExecutionResponse do
+  @moduledoc """
+  Execution item response API spec
+  """
+
+  require OpenApiSpex
+
+  alias OpenApiSpex.Schema
+
+  alias WandaWeb.Schemas.ExecutionResponse.CheckResult
+
+  OpenApiSpex.schema(%{
+    title: "ExecutionResponse",
+    description: "The representation of an execution, it may be a running or completed one",
+    type: :object,
+    properties: %{
+      execution_id: %Schema{type: :string, format: :uuid, description: "Execution ID"},
+      group_id: %Schema{type: :string, format: :uuid, description: "Group ID"},
+      status: %Schema{
+        type: :string,
+        enum: ["running", "completed"],
+        description: "The status of the current execution"
+      },
+      started_at: %Schema{
+        type: :string,
+        format: :"date-time",
+        description: "Execution start time"
+      },
+      completed_at: %Schema{
+        type: :string,
+        nullable: true,
+        format: :"date-time",
+        description: "Execution completion time"
+      },
+      result: %Schema{
+        type: :string,
+        enum: ["passing", "warning", "critical", "unknown"],
+        description: "Aggregated result of the execution, unknown for running ones"
+      },
+      timeout: %Schema{
+        type: :array,
+        items: %Schema{type: :string, format: :uuid, description: "Agent ID"},
+        description: "Timed out agents"
+      },
+      check_results: %Schema{type: :array, items: CheckResult}
+    },
+    required: [
+      :execution_id,
+      :group_id,
+      :status,
+      :started_at,
+      :completed_at,
```

Comment:
I'm not sure on open api `required` usage. Does this require to have a populdated field? I mean, for example `completed_at` and `timeout` can be empty values

Reply:
Can be either empty lists or null, but they will always be there, for simplicity. Both in running and completed executions.
I can also remove something from the required list, what about the actual api then? would it return `completed_at: null` for a running execution or would it not return it at all?

Reply:
return null for every not yet populated field works for me